src/utils/parse-request.js

```diff
@@ -5,15 +5,15 @@ export default function parseRequest(request = {}, cmsUrls) {
     cmsUrls = globalCmsUrls;
   }
 
-  const urlPath = request.path;
+  const [urlPath, query = ''] = request.path.split('?', 2);
   let hostname = request.hostname;
   // remove port number from hostname
   if (hostname.indexOf(':') != -1) {
     hostname = hostname.substring(0, hostname.indexOf(':'));
   }
 
   // detect if in CMS/preview mode
-  let preview = hasPreviewQueryParameter(urlPath);
+  let preview = hasPreviewQueryParameter(query);
```

Comment:
`hasPreviewQueryParameter` checks for the existence of '?', but we're passing the query itself now which won't contain the starting '?' anymore. That seems to be a bug. Either the function parameter of `hasPreviewQueryParameter` has to be changed to 'query' too (and the code should not assume a leading '?') or the urlPath should be passed.

Reply:
Good finding, I've fixed that by updating `hasPreviewQueryParameter` implementation. Thanks!